cpp/arcticdb/codec/core.hpp

```diff
@@ -283,4 +288,57 @@ struct GenericBlockEncoder {
     }
 };
 
+/// @brief Encode a block based on encoding options.
+/// Just as arcticdb::detail::struct GenericBlockEncoder this class does not care about the specific content of the
+/// block and does not take advantage of specific patterns of data. The main difference between this and
+/// arcticdb::detail::struct GenericBlockEncoder is that the latter stores the shapes data at the beginning of each
+/// block, which is suboptimal. arcticdb::detail::GenericBlockEncoder2 does not care about dimensionality and
+/// does not encode the shapes of the block. For more information see comment above arcticdb::ColumnEncoder2
+template<class BlockType, class TD, class EncoderType>
```

Comment:
Should be at least V2, or ideally a behaviour-modifying tag type as mentioned above.